include/pistache/client.h

```diff
@@ -126,6 +233,7 @@ struct Connection : public std::enable_shared_from_this<Connection> {
     };
 
     std::atomic<uint32_t> state_;
+    std::mutex connectionStateMutex_;
```

Comment:
I would suggest make it mutable. This alow function `isConnected` still be const.

Reply:
Agreed, done